app/messages/send-notification.js

```diff
@@ -0,0 +1,26 @@
+import crypto from 'crypto'
+import { NotifyClient } from 'notifications-node-client'
+
+const sendNotification = async (message) => {
+  const notifyClient = new NotifyClient(process.env.NOTIFY_API_KEY)
+
+  const emailAddresses = Array.isArray(message.body.data.commsAddress)
+    ? message.body.data.commsAddress
+    : [message.body.data.commsAddress]
+
+  for (const emailAddress of emailAddresses) {
+    try {
+      await notifyClient.sendEmail(
+        message.body.data.notifyTemplateId,
+        emailAddress, {
+          personalisation: message.body.data.personalisation,
+          reference: crypto.randomUUID()
+        }
+      )
+    } catch (error) {
+      throw new Error('Error sending email: ', error)
```

Comment:
I'd maybe just log rather than throw.  If you throw you'll stop other email addresses from being sent.

Throwing here will cause the service bus message to be redelivered.  So if there's an issue with one email address, the other recipients will get 2000 emails.

We probably should think about this during the retry work to see how best to handle individual issues.  Once we've got validation in place that should prevent email issues having such an impact too as none will get this far.

Reply:
Done.